File: solo/puzzles/easy/power-of-thor-episode-1/power_of_thor_episode_1.rs
use std::io;

macro_rules! parse_input {
   ($x:expr, $t:ident) => ($x.trim().parse::<$t>().unwrap())
}

fn main() {
   let mut input_line = String::new();
   io::stdin().read_line(&mut input_line).unwrap();
   let inputs = input_line.split(" ").collect::<Vec<_>>();
   let light_x = parse_input!(inputs[0], i32); // the X position of the light of power
   let light_y = parse_input!(inputs[1], i32); // the Y position of the light of power
   let initial_tx = parse_input!(inputs[2], i32); // Thor's starting X position
   let initial_ty = parse_input!(inputs[3], i32); // Thor's starting Y position
   let mut tx = initial_tx;
   let mut ty = initial_ty;
   loop {
      let mut input_line = String::new();
      io::stdin().read_line(&mut input_line).unwrap();
      let remaining_turns = parse_input!(input_line, i32); // The remaining amount of turns Thor can move. Do not remove this line.

      if (light_y < ty) {
         print!("N");
         --ty;
      } else if (light_y > ty) {
         print!("S");
         ty += 1;
      }

      if (light_x < tx) {
         print!("W");
         --tx;
      } else if (light_x > tx) {
         print!("E");
         tx += 1;
      }
      println!();
   }
}

File: solo/puzzles/easy/power-of-thor-episode-1/power_of_thor_episode_1.java
import java.util.*;
import java.io.*;
import java.math.*;

class Player {

   public static void main(String args[]) {
      Scanner in = new Scanner(System.in);
      int lightX = in.nextInt(); // the X position of the light of power
      int lightY = in.nextInt(); // the Y position of the light of power
      int initialTx = in.nextInt(); // Thor's starting X position
      int initialTy = in.nextInt(); // Thor's starting Y position
      int tx = initialTx;
      int ty = initialTy;
      // game loop
      while (true) {
         int remainingTurns = in.nextInt(); // The remaining amount of turns Thor can move. Do not remove this line.

         if (lightY < ty)
         {
            System.out.print("N");
            --ty;
         }
         else if (lightY > ty)
         {
            System.out.print("S");
            ++ty;
         }

         if (lightX < tx)
         {
            System.out.print("W");
            --tx;
         }
         else if (lightX > tx)
         {
            System.out.print("E");
            ++tx;
         }
         System.out.println("");
      }
   }
}
